### PR TITLE
[UBS.Admin.Permissions] API: Admin is able to edit SuperAdmin's permissions #5554

### PR DESCRIPTION
**Pull request by ABbondar**
# GreenCityUser PR
https://github.com/ita-social-projects/GreenCity/issues/5554

## Summary Of Changes :fire:
Added validation for updating permissions for employee.
Added several new endpoints and tests.

## Issue Link :clipboard:
_&lt;Link to the issue&gt;_

## Added
* Added validation for updating employee's permissions.
* Added new endpoint for getting information about login employee`s positions.
* Added new endpoint for geting information about an employee`s positions and all possible related authorities to these positions.

## Changed
* _&lt;detail item of what was changed&gt;_

## Deleted
* _&lt;detail item of what was removed&gt;_

## How to test :clipboard:
In can be test via swagger:
[/admin/ubs-employee/edit-authorities]
[/admin/ubs-employee/get-employee-login-positions]
[/admin/ubs-employee/get-positions-authorities]

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: core/src/main/java/greencity/controller/UserController.java [hunk 1]
```diff
@@ -1270,6 +1273,48 @@ public ResponseEntity<Object> getAllAuthorities(@RequestParam String email) {
...
+     * @return List of {@link String} - list of employee positions.
+     *
+     * @author Anton Bondar.
+     */
+    @ApiOperation(value = "Get information about login employee`s positions.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/UserController.java]: please add `404`
you throw `NOT_FOUND` here
https://github.com/ita-social-projects/GreenCityUser/pull/236/files#diff-fb2ad6279cd37f2fdc3daf97cefacf4ba3aced726318d85fdc44509f1be8c628R52

File: core/src/main/java/greencity/controller/UserController.java [hunk 2]
```diff
@@ -1270,6 +1273,48 @@ public ResponseEntity<Object> getAllAuthorities(@RequestParam String email) {
...
+     *
+     * @author Anton Bondar.
+     */
+    @ApiOperation(value = "Get information about an employee`s positions and all possible "
+        + "related authorities to these positions.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/UserController.java]: please add `404`
you throw `NOT_FOUND` here
https://github.com/ita-social-projects/GreenCityUser/pull/236/files#diff-fb2ad6279cd37f2fdc3daf97cefacf4ba3aced726318d85fdc44509f1be8c628R30

File: core/src/main/java/greencity/controller/UserController.java (hunk 1 above)
Comment (ABbondar) [reply core/src/main/java/greencity/controller/UserController.java]: added

File: core/src/main/java/greencity/controller/UserController.java (hunk 2 above)
Comment (ABbondar) [reply core/src/main/java/greencity/controller/UserController.java]: added

**Review by RomaMocherniuk — APPROVED** [review]

**Review by MaksymGolik — APPROVED** [review]

**Review by LiliaMokhnatska — APPROVED** [review]

**Review by sergoliarnik — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Pull request merged**